docs/pages/guides/infrastructure-management/how-meshery-manages-deployments.md

```diff
@@ -0,0 +1,275 @@
+
+### 2.2 Versioned Entity Lifecycle
+
+In Meshery, every component — from a service to a CRD — follows a **versioned lifecycle**.  
+This helps ensure changes are:
+
+- Traceable
+- Repeatable
+- Compatible across environments
+
+[![Meshery Entities]({{ site.baseurl }}/assets/img/infrastructure-management/meshery-entities.svg)]({{ site.baseurl }}/assets/img/infrastructure-management/meshery-entities.svg)
+
+#### Understanding Meshery’s Versioning Model
+
+Each Meshery entity follows a structured versioning lifecycle, ensuring consistency before deployment and management. This process enables users to track changes, apply upgrades, and maintain compatibility across environments.
+
+- **Schema** (`v1alpha1`): Defines the fundamental structure of an entity, ensuring consistency and validation.  
+- **Definition**: Establishes the expected behavior of an entity, detailing constraints and relationships.  
+- **Declaration**: Provides a human-readable YAML-based configuration that Meshery uses for deployment.  
+- **Instance**: Represents the actual deployed resource, tracking runtime changes and operational metadata.  
+
+## 3. Core Operation Logic
+
+Once a model is ready, Meshery takes care of deploying it into your Kubernetes cluster and tracking its lifecycle.
+
+This section breaks down what happens during a **Deploy** or **Undeploy** action — how Meshery applies changes, handles dependencies, and updates its internal state.
+
+### 3.1 Deploy Action Breakdown
+
+Meshery follows a consistent workflow when you trigger a deployment. The goal is to ensure the process is safe, dependency-aware, and fully observable.
+
+[![Deploy Action Flow]({{ site.baseurl }}/assets/img/infrastructure-management/meshery-deploy-action.svg)]({{ site.baseurl }}/assets/img/infrastructure-management/meshery-deploy-action.svg)
+
+#### What happens during deployment:
+
+1. **Parse Design File**  
+   Meshery reads the `meshery-design.yaml` file, which describes what should be deployed and how it’s structured.
+2. **Validate Dependencies**  
+   Meshery checks for required CRDs, operators, and supporting components before starting. If something’s missing, it may install it or alert you.
+3. **Deploy via Adapter**  
+   Each deployment request is passed to the appropriate Meshery Adapter (e.g., Istio, Linkerd) via gRPC. The Adapter translates the model into actual Kubernetes resources.
+4. **Sync with MeshSync**  
+   Meshery continuously tracks the deployed resources via **MeshSync** to monitor their status in real time.
+
+#### Error Handling and Recovery
+- If a dependency is missing, Meshery will either attempt to install it or notify the user.
+- Failures during deployment are logged and surfaced through the UI and CLI.
```

Comment:
Should we mention that UI is Kanvas and CLI is mesheryctl? (For readers who come to read this document first and may not be aware of what these are)